src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.Iterator.cs

```diff
@@ -10,11 +11,11 @@ internal partial class TokenStream
     {
         // gain of having hand written iterator seems about 50-100ms over auto generated one.
         // not sure whether it is worth it. but I already wrote it to test, so going to just keep it.
-        private class Iterator : IEnumerable<ValueTuple<int, SyntaxToken, SyntaxToken>>
+        private struct Iterator : IEnumerable<ValueTuple<int, SyntaxToken, SyntaxToken>>
```

Comment:
is it valuable to implement this interface (since it would box)?  should we actually get rid of it to make sure the consumer doesn't ever box?